more_itertools/more.py

```diff
@@ -741,14 +741,37 @@ def _partial(A, r):
             i += 1
             head[i:], tail[:] = tail[: r - i], tail[r - i :]
 
-    items = sorted(iterable)
+    try:
+        items = sorted(iterable)
+        sortable = True
+    except TypeError:
+        items = list(iterable)
+        sortable = False
+
+        # Sets are unordered.  Could use list(dict.fromkeys(items))
+        # if every item was Hashable.
+        # This is suboptimal, but both functions above are already O(n^2)
```

Comment:
What's O(n^2)? The permuting functions? Certainly not, as there are usually more than quadratically many permutations.

Reply:
_full and _partial.  I've not looked at them in detail, I just noticed they both contained 2 or 3 for loops (in sequence) within a while loop.

I can change it to "at best O(n^2)" if you like?  The point of that comment still stands:  using O(n) sequential look up instead of an O(1) hash table for the n items, does not make things subtantially slower than it would be already

Reply:
I'd say "are already worse than O(n^2)".

Reply:
Done.